x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtValidateUtil.java

```diff
@@ -286,17 +280,17 @@ public static void validateSignature(final SignedJWT jwt, final List<JWK> jwks)
         assert jwks != null && jwks.isEmpty() == false : "Caller must provide a non-empty JWK list";
         final String id = jwt.getHeader().getKeyID();
         final JWSAlgorithm alg = jwt.getHeader().getAlgorithm();
-        LOGGER.trace("JWKs [" + jwks.size() + "], JWT KID [ + id + ], and JWT Algorithm [" + alg.getName() + "] before filters.");
+        LOGGER.trace("JWKs [{}], JWT KID [{}], and JWT Algorithm [{}] before filters.", jwks.size(), id, alg.getName());
 
         // If JWT has optional kid header, and realm JWKs have optional kid attribute, any mismatches JWT.kid vs JWK.kid can be ignored.
         // Keep any JWKs if JWK optional kid attribute is missing. Keep all JWKs if JWT optional kid header is missing.
         final List<JWK> jwksKid = jwks.stream().filter(j -> ((id == null) || (j.getKeyID() == null) || (id.equals(j.getKeyID())))).toList();
-        LOGGER.trace("JWKs [" + jwksKid.size() + "] after KID [" + id + "||null] filter.");
+        LOGGER.trace("JWKs [{}] after KID [{}](|null) filter.", jwksKid.size(), id);
 
         // JWT has mandatory alg header. If realm JWKs have optional alg attribute, any mismatches JWT.alg vs JWK.alg can be ignored.
         // Keep any JWKs if JWK optional alg attribute is missing.
         final List<JWK> jwksAlg = jwksKid.stream().filter(j -> (j.getAlgorithm() == null) || (alg.equals(j.getAlgorithm()))).toList();
-        LOGGER.trace("JWKs [" + jwksAlg.size() + " after Algorithm [" + alg.getName() + "||null] filter.");
+        LOGGER.trace("JWKs [{}] after Algorithm [{}](|null) filter.", jwksAlg.size(), alg.getName());
```

Comment:
Nit. This is more than a parameterization change, content of the message changed too. It is not clear if that was intended. If intended then OK.
- Before: `Algorithm [HS256||null]`
- After: `Algorithm [HS256](|null)`

Reply:
It is intended - sorry, I should have noted it before review.
The `[]` are intended to show where a parameters have been inserted, so we can clearly distinguish things like stray whitespace that could be causing a problem. Hence it's not intended that static text (like `||null`) be inside those brackets.  
I figured this is one of the messges we might revisit later, so I made only minor tweaks, but thought it was worth bringing it inline with the standard way we output messages.